### PR TITLE
Upgrade presto pinot connector to be compatible with new Broker routing and time boundary APIs in Pinot 0.3.0 release

### PR DESCRIPTION
**Pull request by fx19880617**
Recently Pinot upgraded the Broker side routing table API and changed the behavior of timeBoundary query API. This PR aims to be compatible with new changes.

```
== RELEASE NOTES ==

Pinot Changes
* Adding support for new Pinot Routing Table APIs.
```

### CODE REVIEW

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java [hunk 1]
```diff
@@ -359,10 +395,8 @@ public RoutingTables(@JsonProperty("routingTableSnapshot") List<RoutingTableSnap
             else {
                 List<Map<String, List<String>>> routingTableEntriesList = snapshot.getRoutingTableEntries();
                 if (routingTableEntriesList.isEmpty()) {
-                    throw new PinotException(
-                            PINOT_UNEXPECTED_RESPONSE,
-                            Optional.empty(),
-                            String.format("Empty routingTableEntries for %s. RoutingTable: %s", tableName, responseBody));
+                    throw new PinotException(PINOT_UNEXPECTED_RESPONSE, Optional.empty(),
+                        String.format("Empty routingTableEntries for %s. RoutingTable: %s", tableName, responseBody));
```
Comment (zhenxiao) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java]: either put all parameters in one line, or one in each line

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java [hunk 2]
```diff
@@ -426,7 +460,21 @@ public TimeBoundary(
...
+            String responseBody = sendHttpGetToBroker(table, String.format(TIME_BOUNDARY_API_TEMPLATE, table));
+            return timeBoundaryJsonCodec.fromJson(responseBody);
+        }
+        catch (Exception e) {
+            if ((e instanceof PinotException)) {
+                // New Pinot broker will set response code 404 if time boundary is not set.
+                // This is backward incompatible with old version, as it returns an empty json object.
+                // In order to gracefully handle this, below check will extract response code and return empty json
+                // if not found time boundary.
+                if (e.getMessage().split(" ")[3].equalsIgnoreCase("404")) {
```
Comment (zhenxiao) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java]: make "404" a constant?
check and guard no array index out of bound exception when accessing [3]? could you please comment error message format

Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java]: Done

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java (hunk 1 above)
Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java]: done

**Comment by fx19880617** [comment]
> nice work, @fx19880617
> looks good, just 2 minor issues

Thanks @zhenxiao ! Updated the PR to address your comments.

File: presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java [hunk 3]
```diff
@@ -426,7 +463,26 @@ public TimeBoundary(
...
+            if ((e instanceof PinotException)) {
+                /** New Pinot broker will set response code 404 if time boundary is not set.
+                 * This is backward incompatible with old version, as it returns an empty json object.
+                 * In order to gracefully handle this, below check will extract response code and return empty json
+                 * if not found time boundary.
+                 *
+                 * Sample error message:
+                 *     Unexpected response status: 404 for request  to url http://127.0.0.1:8000/debug/timeBoundary/baseballStats, with headers ...
+                 */
+                String[] errMsgSplits = e.getMessage().split(" ");
```
Comment (zhenxiao) [inline presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java:480]: s/errMsgSplits/errorMessageSplits/g
minor thing, Presto usually not use abbreviation in variable name

Comment (fx19880617) [reply presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java]: Sure! Still getting used to Presto coding style :)

**Pull request merged**